kubernetes/elastic-helm/daemonset.yaml

```diff
@@ -1,56 +0,0 @@
-mode: daemonset
```

Comment:
Does this mean that we only use the `Deployment`?
If so, will that work with a multi-node cluster?
The purpose of the Daemonset is to ensure that node level metrcis+logs are collected from each node by the respective Collector.

Reply:
Great catch! I assumed we were already deploying it as a daemonset. I think that as we are enabling the `hostmetrics` and `filelog` receiver the default mode should be `daemonset` and remove the deployment one.

Mode changed in https://github.com/elastic/opentelemetry-demo/pull/33/commits/80612a5a2ceb022b50975b68ab035eda9b8c04c3

Reply:
But then we will be collecting `k8scluster` metrics and events from all the nodes, right? Unless we want to remove them for some reason?

In general, the Collector is recommended to be deployed in both modes at the same time and each Resource handels its own part. The Daemonset takes care of the node level collection and the Deployment acts a singleton collecting the cluster level signals (metrics but traces from the SDKs as well).
Or we don't want to achieve this here?

Reply:
> But then we will be collecting k8scluster metrics and events from all the nodes, right? Unless we want to remove them for some reason?

k8scluster metrics are disabled when deployed as a deamonset: https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-collector#configuration-for-kubernetes-cluster-metrics

> In general, the Collector is recommended to be deployed in both modes at the same time and each Resource handels its own part.

Makes sense to me. If it sounds good to you, I will rollback the removal of the `daemonset` file and document the following two deployment modes:

- Deployment: Otlp receiver + k8scluster metrics + Otlp exporter
- Daemonset: Hostmetrics + filelogreceiver (and its processors) + elasticsearch exporter

My main concern would be with the logs, the demo services send Otlp logs that can be either gathered by the Otlp receiver or the filelog receiver. Should we exclude those from being collected with the filelog receiver to prevent duplications?

Reply:
> Deployment: Otlp receiver + k8scluster metrics + Otlp exporter
> Daemonset: Hostmetrics + filelogreceiver (and its processors) + elasticsearch exporter

That sounds good! For the Daemonset all the receivers that make sense on Node scope and the Deployment everything that is cluster scope (k8scluster metrics, k8s events etc).

For the logs duplication one thing we can do is to apply the [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md) to drop logs collected by the filelog receiver in case some k8s labels apply? Or based on a similar pattern, we can annotate the instrumented apps with a specific annotation to indicate that these should be dropped.

Reply:
> For the logs duplication one thing we can do is to apply the [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md) to drop logs collected by the filelog receiver in case some k8s labels apply? Or based on a similar pattern, we can annotate the instrumented apps with a specific annotation to indicate that these should be dropped.

The issue is that not all the Otel demo services have been migrated to Otel logs, instead some output to stdout and smothers to an Otel logs provider. See tracking issue: https://github.com/open-telemetry/opentelemetry-demo/issues/1472.

This is an important topic to tackle, but I think it would make sense to move the discussion and implementation to its own issue/pr, wdyt?